### DISCUSSION TITLE
How enable/add custom attributes/enable any attribute?

### DISCUSSION THREAD

**Discussion by wintercounter** [JiLiZART/BBob#47] [Q&A] [answered]
I'm trying to figure out from source code but couldn't really get where and how to pass options to make this work, the docs are not so imformative about this.

At the company we have our own BBCode syntax. We allow attributes with `@` sign, eg: `[b @class="myClass"]Text[/b]`. We support basically any attribute. I want to enable all attributes that's being passed and also add suport to our custom ones somehow.

**Comment by JiLiZART** [comment, marked as answer]
Hi, @wintercounter you can extend default html5 preset. 

```js
import html5Preset from "@bbob/preset-html5";
import toHTML from "@bbob/html";

const myPreset = html5Preset.extend(tags => ({
  ...tags,
  b: (node, params) => ({
    ...tags.b(node, params),
    attrs: {
      style: "font-weight: bold;"
    },
    content: JSON.stringify(node.attrs)
  })
}));

document.body.innerHTML = toHTML(`[b @class="myClass"]Text[/b]`, [myPreset()]);
```
Demo example
https://codesandbox.io/embed/blue-water-l0pc6
